api/src/passport.setup.js

```diff
@@ -42,11 +38,23 @@ passport.use(
           name: profile.displayName,
           email: profile.emails[0].value,
           profileImg: profile.photos[0].value,
-        },
-      });
-      //console.log('big access token: ', accessToken);
-      const token = jwt.sign({ accessToken, refreshToken, googleKey: profile.id }, process.env.JWT_SECRET);
-      return done(null, token);
+        }
+      })
+        .then((user) => {
+          if (!user[0].calendarId) {
+            console.log('making calendar for: ', user[0]);
+            let calendar = Calendar.createCalendar(accessToken, refreshToken);
+            Promise.resolve(calendar)
+              .then((calendarObject) => {
```

Comment:
While this definitely works, you should be able to just do: 
```
Calendar.createCalendar(accessToken, refreshToken)
.then((calendarObject) => {
//...
```
I'm curious if there's a specific reason you didn't do it this way.

Reply:
I have refactored this code to save those few lines.